lua/telescope/make_entry.lua

```diff
@@ -453,6 +453,34 @@ function make_entry.gen_from_marks(_)
   end
 end
 
+function make_entry.gen_from_registers(_)
+  local displayer = entry_display.create {
+    separator = ":",
+    items = {
+      { width = 4 },
+      { remaining = true },
+    },
+  }
+
+  local make_display = function(register, content)
+    return displayer {
+      string.format("[%s]", register.key),
+      content,
+    }
+  end
+
+  return function(entry)
+    local reg_content = vim.fn.getreg(entry.key)
+    return {
+      value = entry.key,
+      ordinal = entry.key,
+      content = reg_content,
+      regtype = vim.fn.getregtype(entry.key),
+      display = make_display(entry, reg_content),
```

Comment:
You should not all make_entry. The picker does that on demand which is better for performance. The `entry` which you will get would be yourself. So in make entry `register` becomes `entry` and `content` would be `entry.content`.

Reply:
Reading over this makes no sense. What i mean is we don't call `make_display` when we return this table. What you actually want to do is just do `display = make_display` and your `make_display` function signature would be just `(entry)`, which is the table, that we return. `key` would be `entry.value` and `content` would be `entry.content`